src/python/pants/backend/python/lint/ruff/subsystem.py

```diff
@@ -30,17 +40,43 @@ class RuffMode(str, Enum):
 # --------------------------------------------------------------------------------------
 
 
-class Ruff(PythonToolBase):
+class Ruff(TemplatedExternalTool):
     options_scope = "ruff"
     name = "Ruff"
-    help_short = "The Ruff Python formatter (https://github.com/astral-sh/ruff)."
-
-    default_main = ConsoleScript("ruff")
-    default_requirements = ["ruff>=0.1.2,<1"]
-
-    register_interpreter_constraints = True
-
-    default_lockfile_resource = ("pants.backend.python.lint.ruff", "ruff.lock")
+    help = "The Ruff Python formatter (https://github.com/astral-sh/ruff)."
+
+    default_version = "0.6.4"
```

Comment:
Note - latest is now 0.7.1 but I think we can upgrade that in a follow-up rather than adding to this PR.

Reply:
Yeah, agreed

Reply:
https://github.com/pantsbuild/pants/pull/21605